Alicerce/Sources/Logging/Log.swift

```diff
@@ -37,55 +37,61 @@ public final class Log {
     public class func verbose(_ message: @autoclosure () -> String,
                               file: String = #file,
                               function: String = #function,
-                              line: Int = #line) {
+                              line: Int = #line,
+                              completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
```

Comment:
I'm not sure about exposing these `completion`'s as an API for the logging caller.

I think that we should do some sort of error handling on the destinations, but I think exposing this to the outside  via closures is probably not the best option.

Since LogDestinations should write asynchronously, it doesn't make sense to make `write` throw, so we have to somehow propagate possible errors upstream to the LogStack.

My suggestion would be to either define a Delegate for the destination, or simply an `errorClosure` var on the Destinations that the LogStack sets and handles any error that occurs on a given log. With this, the LogStack itself can also provide a similar API so that the LogStack user can react to possible errors.

So, in each Destination instead of having to always call a completion for each logItem we would simply invoke the error delegate or the errorClosure when something bad occurs.

What do you say?